crates/core/src/sync_local.rs

```diff
@@ -288,19 +291,35 @@ GROUP BY b.row_type, b.row_id",
     }
 
     fn mark_completed(&self) -> Result<(), SQLiteError> {
-        if self.partial.is_none() {
-            // language=SQLite
-            self.db
-                .exec_safe("DELETE FROM ps_updated_rows")
-                .into_db_result(self.db)?;
+        let priority_code = match &self.partial {
+            None => {
+                // language=SQLite
+                self.db
+                    .exec_safe("DELETE FROM ps_updated_rows")
+                    .into_db_result(self.db)?;
+                -1
+            }
+            Some(partial) => partial.priority.into(),
+        };
 
-            // language=SQLite
-            self.db
-                .exec_safe(
-                    "insert or replace into ps_kv(key, value) values('last_synced_at', datetime())",
-                )
-                .into_db_result(self.db)?;
-        }
+        // Higher-priority buckets are always part of lower-priority sync operations too, so we can
+        // delete information about higher-priority syncs (represented as lower priority numbers).
+        // A complete sync is represented as -1.
```

Comment:
The `-1` here makes the sort ordering a bit weird. You essentially have:

```
0: synced priority 0
1: synced priority 0 + 1
2: synced priority 0 + 1 + 2
3: synced priority 0 + 1 + 2 + 3
-1: synced priority 0 + 1 + 2 + 3
```

An example if this having in effect is in the delete query below (`(priority < ?1) OR (?1 = -1)`).

Could we perhaps use a constant for this that is actually bigger than the other priority values?

Reply:
I've added `i32::MAX` as a priority representing a complete sync.